pkg/providers/snow/apibuilder.go

```diff
@@ -26,12 +27,16 @@ func CAPICluster(clusterSpec *cluster.Spec, snowCluster *snowv1.AWSSnowCluster,
 	return clusterapi.Cluster(clusterSpec, snowCluster, kubeadmControlPlane)
 }
 
-func KubeadmControlPlane(clusterSpec *cluster.Spec, snowMachineTemplate *snowv1.AWSSnowMachineTemplate) (*controlplanev1.KubeadmControlPlane, error) {
+func KubeadmControlPlane(clusterSpec *cluster.Spec, snowMachineTemplate *snowv1.AWSSnowMachineTemplate, log logr.Logger) (*controlplanev1.KubeadmControlPlane, error) {
```

Comment:
nit: 
```suggestion
func KubeadmControlPlane(log logr.Logger, clusterSpec *cluster.Spec, snowMachineTemplate *snowv1.AWSSnowMachineTemplate) (*controlplanev1.KubeadmControlPlane, error) {
```
For consistency, we usually pass the logger before the other arguments